plugins/home/src/components/CustomHomepage/CustomHomepageGrid.tsx

```diff
@@ -0,0 +1,363 @@
+        borderColor: `transparent transparent ${theme.palette.primary.light} transparent`,
+      },
+    },
+    contentHeaderBtn: {
+      marginLeft: theme.spacing(2),
+    },
+    widgetWrapper: {
+      '& > *:first-child': {
+        width: '100%',
+        height: '100%',
+      },
+      '& + .react-grid-placeholder': {
+        backgroundColor: theme.palette.primary.light,
+      },
+      '&.edit > :active': {
+        cursor: 'move',
+      },
+    },
+  }),
+);
+
+function useHomeStorage(
+  defaultWidgets: GridWidget[],
+): [GridWidget[], (value: GridWidget[]) => void] {
+  const key = 'home';
+  const storageApi = useApi(storageApiRef).forBucket('home.customHomepage');
+  // TODO: Support multiple home pages
+  const setWidgets = useCallback(
+    (value: GridWidget[]) => {
+      const grid: CustomHomepageGridStateV1 = {
+        version: 1,
+        pages: {
+          default: value,
+        },
+      };
+      storageApi.set(key, JSON.stringify(grid));
+    },
+    [key, storageApi],
+  );
+  const homeSnapshot = useObservable(
+    storageApi.observe$<string>(key),
+    storageApi.snapshot(key),
+  );
+  const widgets: GridWidget[] = useMemo(() => {
+    if (homeSnapshot.presence === 'absent') {
+      return defaultWidgets;
+    }
+    try {
+      const grid: CustomHomepageGridStateV1 = JSON.parse(homeSnapshot.value!);
+      return CustomHomepageGridStateV1Schema.parse(grid).pages.default;
+    } catch (e) {
+      return defaultWidgets;
+    }
+  }, [homeSnapshot, defaultWidgets]);
+
+  return [widgets, setWidgets];
+}
+
+const convertConfigToDefaultWidgets = (
+  config: LayoutConfiguration[],
+  availableWidgets: Widget[],
+): GridWidget[] => {
+  const ret = config.map((conf, i) => {
+    const c = LayoutConfigurationSchema.parse(conf);
+    const name = getComponentData(c.component, 'core.name');
+    if (!name) {
+      return null;
+    }
+    const widget = availableWidgets.find(w => w.name === name);
+    if (!widget) {
+      return null;
+    }
+    const widgetId = `${widget.name}__${i}${Math.random()
+      .toString(36)
+      .slice(2)}`;
+    return {
+      id: widgetId,
+      layout: {
+        i: widgetId,
+        x: c.x,
+        y: c.y,
+        w: Math.min(widget.maxWidth ?? Number.MAX_VALUE, c.width),
+        h: Math.min(widget.maxHeight ?? Number.MAX_VALUE, c.height),
+        minW: widget.minWidth,
+        maxW: widget.maxWidth,
+        minH: widget.minHeight,
+        maxH: widget.maxHeight,
+        isDraggable: false,
+        isResizable: false,
+      },
+      settings: {},
+    };
+  });
+  return compact(ret);
+};
+
+const availableWidgetsFilter = (elements: ElementCollection) => {
+  return elements
+    .selectByComponentData({
+      key: 'core.name',
+    })
+    .getElements<Widget>()
+    .flatMap(elem => {
+      const cardLayout = getComponentData<CardLayout>(elem, 'layout');
+      const cardSettings = getComponentData<CardSettings>(elem, 'settings');
```

Comment:
Let's collect all of these under a single or a few keys instead, something like `home.widget.config`. I'd like to avoid too generic keys, and also too many keys as that can make it quite messy to evolve this over time.

Reply:
Ok, could you make up your mind about the naming? I have changed these already 3 or more times according to comments :D I am going to leave on vacation today so changing these will go to the unknown future, maybe after 2-3 weeks if nobody else has time to contribute to this PR.

Reply:
The naming isn't what I'm after here really though, it's about grouping the component data up under one or a few keys

Reply:
Moved these under `home.widget.config`. The `createCardExtensions` supports adding these as previously but the data is hidden behind that key. If users use the `createComponentExtension` instead, they have to set it manually to the data object (see README.md)